go/libraries/doltcore/sqle/cluster/controller.go

```diff
@@ -64,6 +64,15 @@ const RoleDetectedBrokenConfig Role = "detected_broken_config"
 
 const PersistentConfigPrefix = "sqlserver.cluster"
 
+// State for any ongoing DROP DATABASE replication attempts we have
+// outstanding. When we create a database, we cancel all on going DROP DATABASE
+// replication attempts.
+type databaseDropReplication struct {
```

Comment:
I think ideally a different approach would be possible, but I don't think it's particularly near at hand. The current approach is quite near at hand and is probably good enough. In particular:

1. I considered blocking or failing creation until the drop was fully replicated. But it's not particularly compelling because the replica can be down for an arbitrary amount of time, in which case you're needlessly taking an availability hit on the primary. There doesn't seem to be a compelling reason to do this when it's easy enough to just cancel the drop database replication.
2. A queue of replication events isn't really what you want. When a replica comes back online, you want to true it up to whatever the current state is, without going through all the intermediate steps which: (a) are arbitrarily large, (b) are irrelevant to any given client and are ideally never seen by a client, (c) are more expensive to enact than just replicating the current state.

The presented approach basically shoots for having at most one ongoing replication attempt for a given database name. Either we're not replicating anything for that name, or we're replicating a delete, or we're replicating a new head. An alternative approach which I considered was to put the delete replication itself into the commithook, and to have the commit hook remove itself, or something like that, once it was successfully replicated, and to have the initdatabasehook cancel those things before adding the new commit hooks (or just set existing commit hooks back to replicating the new head...). It would look somewhat similar, but also somewhat different. commithook is currently quite large and quite subtle, so I decided to forego adding even more complications there for the time being and keep the drop database replication outside of it for now.